src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4329,6 +4329,12 @@ private BoundExpression MakeBadExpressionForObjectCreation(ObjectCreationExpress
             {
                 var memberName = (IdentifierNameSyntax)namedAssignment.Left;
 
+                var allInitializerExpressionsAreIndexExpressions =
```

Comment:
Yeah, this looks wrong. If my read of the spec is correct, the categorization of the receiver is non-moveable. So we shouldn't need to know if these are index expressions in the example

```C#
using System;

unsafe struct S
{
    public fixed double V[3];

    static void Main()
    {
        var s = new S { V = 
        {
            [0] = 0,
            [1] = 1,
            [2] = 2,
        } };
        
        Console.Write(s.V[0]);
        Console.Write(s.V[1]);
        Console.Write(s.V[2]);
    }
}
```

I think the problem may be in `Binder.IsMoveableVariable`. Initializers should create a BoundObjectOrCollectionValuePlaceholder and, by my read, that placeholder (as long as it is a struct) is always a local temporary and thus immoveable.